smarts/core/sumo_road_network.py

```diff
@@ -193,6 +193,9 @@ def edge_by_id(self, edge_id):
     def lane_by_id(self, lane_id):
         return self._graph.getLane(lane_id)
 
+    def lane_by_offset_on_edge(self, edge_id, lane_idx):
```

Comment:
lane offset means something very specific elsewhere in the code, namely, the distance along the lane from start of the lane. We shouldn't mix these meanings with indices.

Try `lane_by_index_on_edge`